shared/workspace_repo/api_client.go

```diff
@@ -192,7 +240,7 @@ func (c *WorkspaceRepo) Stacks() ([]string, error) {
 }
 
 func (c *WorkspaceRepo) GetWorkspace(ctx context.Context, workspaceName string) (Workspace, error) {
-	client, err := c.getTfc(ctx)
+	client, err := c.getTfcSync()
```

Comment:
This operation needs to be threadsafe so multiple goroutines can get the same client and validation only needs to happen one time. A new function was made to indicate it is a synchronous operation.